### PR TITLE
Improve tests for Labels

### PR DESCRIPTION
**Pull request by res0nance**
Improve testing for labels as the code had no tests.

### Proposed changelog entries

* Entry 1: Minor
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: test/src/test/java/hudson/model/labels/LabelAtomTest.java [hunk 1]
```diff
@@ -20,5 +26,57 @@ public void selfLabel() throws Exception {
...
+        Node n2 = j.createSlave("n2", "label label2", null);
+        Node n3 = j.createSlave("n3", "label2", null);
+        Label l = j.jenkins.getLabel("label");
+        Label l2 = j.jenkins.getLabel("label2");
+        Label l3 = j.jenkins.getLabel("label3");
+        assertThat(l.getNodes().size(), is(2));
+        assertThat(l.getNodes(), containsInAnyOrder(n1,n2));
+        assertThat(l2.getNodes().size(), is(2));
+        assertThat(l2.getNodes(), containsInAnyOrder(n3,n2));
+        assertThat(l3.getNodes().isEmpty(), is(true));
```
Comment (MarkEWaite) [inline test/src/test/java/hudson/model/labels/LabelAtomTest.java:45]: Failures may be clearer with
```suggestion
        assertThat(l3.getNodes() is(empty()));
```

File: test/src/test/java/hudson/model/labels/LabelAtomTest.java [hunk 2]
```diff
@@ -20,5 +26,57 @@ public void selfLabel() throws Exception {
...
+
+    @Test
+    public void getClouds() {
+        Cloud test = new TestCloud("test", "label");
+        j.jenkins.clouds.add(test);
+        Label l = new LabelAtom("label");
+        Label l2 = new LabelAtom("label2");
+        assertThat(l.getClouds().size(), is(1));
+        assertThat(l.getClouds(), containsInAnyOrder(test));
+        assertThat(l2.getClouds().isEmpty(), is(true));
```
Comment (MarkEWaite) [inline test/src/test/java/hudson/model/labels/LabelAtomTest.java:56]: Failures may be clearer to the reader with
```suggestion
        assertThat(l2.getClouds(), is(empty()));
```

**Review by MarkEWaite — APPROVED** [review]
Approved.  Optional comments for your consideration (with likely need to add an import)

File: test/src/test/java/hudson/model/labels/LabelAtomTest.java [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
 import org.junit.Rule;
 import org.junit.Test;
 import org.jvnet.hudson.test.JenkinsRule;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
 import static org.hamcrest.Matchers.is;
```
Comment (res0nance) [inline test/src/test/java/hudson/model/labels/LabelAtomTest.java:15]:
```suggestion
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.empty;
```

**Review by MarkEWaite — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by MarkEWaite** [comment]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**